libs/remote_storage/src/s3_bucket.rs

```diff
@@ -427,10 +427,20 @@ impl RemoteStorage for S3Bucket {
     }
 
     async fn download(&self, from: &RemotePath) -> Result<Download, DownloadError> {
+        // if prefix is not none then download file `prefix/from`
+        // if prefix is none then download file `from`
+        let query_key = match &self.prefix_in_bucket {
```

Comment:
WDYT if we move this logic inside `relative_path_to_s3_object` so it can be unit tested? 

Is the fix applicable for other places when `relative_path_to_s3_object` is used?

Reply:
> WDYT if we move this logic inside relative_path_to_s3_object so it can be unit tested?

yes this is a good idea

> Is the fix applicable for other places when relative_path_to_s3_object is used?

AFAIK it should be

Reply:
changed

Reply:
unit tests pass

Reply:
Thanks! Do we have a standalone unit test for `relative_path_to_s3_object`? If not lets add one

Reply:
I added one